pkg/datadog/config/config.go

```diff
@@ -0,0 +1,351 @@
+
+	// Site is the site of the Datadog intake to send data to.
+	// The default value is "datadoghq.com".
+	Site string `mapstructure:"site"`
+
+	// FailOnInvalidKey states whether to exit at startup on invalid API key.
+	// The default value is false.
+	FailOnInvalidKey bool `mapstructure:"fail_on_invalid_key"`
+}
+
+// TagsConfig defines the tag-related configuration
+// It is embedded in the configuration
+type TagsConfig struct {
+	// Hostname is the fallback hostname used for payloads without hostname-identifying attributes.
+	// This option will NOT change the hostname applied to your metrics, traces and logs if they already have hostname-identifying attributes.
+	// If unset, the hostname will be determined automatically. See https://docs.datadoghq.com/opentelemetry/schema_semantics/hostname/?tab=datadogexporter#fallback-hostname-logic for details.
+	//
+	// Prefer using the `datadog.host.name` resource attribute over using this setting.
+	// See https://docs.datadoghq.com/opentelemetry/schema_semantics/hostname/?tab=datadogexporter#general-hostname-semantic-conventions for details.
+	Hostname string `mapstructure:"hostname"`
+}
+
+// Config defines configuration for the Datadog exporter.
+type Config struct {
+	confighttp.ClientConfig      `mapstructure:",squash"` // squash ensures fields are correctly decoded in embedded struct.
+	exporterhelper.QueueSettings `mapstructure:"sending_queue"`
+	configretry.BackOffConfig    `mapstructure:"retry_on_failure"`
+
+	TagsConfig `mapstructure:",squash"`
+
+	// API defines the Datadog API configuration.
+	API APIConfig `mapstructure:"api"`
+
+	// Metrics defines the Metrics exporter specific configuration
+	Metrics MetricsConfig `mapstructure:"metrics"`
+
+	// Traces defines the Traces exporter specific configuration
+	Traces TracesExporterConfig `mapstructure:"traces"`
+
+	// Logs defines the Logs exporter specific configuration
+	Logs LogsConfig `mapstructure:"logs"`
+
+	// HostMetadata defines the host metadata specific configuration
+	HostMetadata HostMetadataConfig `mapstructure:"host_metadata"`
+
+	// OnlyMetadata defines whether to only send metadata
+	// This is useful for agent-collector setups, so that
+	// metadata about a host is sent to the backend even
+	// when telemetry data is reported via a different host.
+	//
+	// This flag is incompatible with disabling host metadata,
+	// `use_resource_metadata`, or `host_metadata::hostname_source != first_resource`
+	OnlyMetadata bool `mapstructure:"only_metadata"`
+
+	// Non-fatal warnings found during configuration loading.
+	warnings []error
+}
+
+// LogWarnings logs warning messages that were generated on unmarshaling.
+func (c *Config) LogWarnings(logger *zap.Logger) {
+	for _, err := range c.warnings {
+		logger.Warn(fmt.Sprintf("%v", err))
+	}
+}
+
+// AddWarning appends to warnings in the configuration.
+func (c *Config) AddWarning(warning error) {
+	c.warnings = append(c.warnings, warning)
+}
```

Comment:
Do we need to make this method public? I want to keep as much of this 'hack' to log warnings private

Reply:
removed